### DISCUSSION TITLE
Priorities of applying specific prices are wrong

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#33747] [Ideas]
### Describe the bug and add attachments

When you set up specific prices like this.

![Snímek obrazovky 2023-08-28 065140](https://github.com/PrestaShop/PrestaShop/assets/6097524/9d485067-5b6e-4cc7-b120-8bac3e1e7f20)

Prestashop sets product price to 90 CZK.

### Expected behavior

It should be 89 CZK IMHO - select the best price the customer can get.

Ping @PrestaShop/product-council

### Steps to reproduce

1. Create a product, set no tax rule, set price to 1000.
2. Create specific price - Visitor group, uncheck "Keep original price" and type 90.
3. Create specific price - All groups, uncheck "Keep original price" and type 89.
4. Visit the product - price is 90, shouldn't it be 89?

### PrestaShop version(s) where the bug happened

8.1

### PHP version(s) where the bug happened

8.1

**Comment by MatShir** [comment]
What is your config priority order?
<img width="1173" alt="Capture d’écran 2023-08-28 à 09 26 06" src="https://github.com/PrestaShop/PrestaShop/assets/43613217/f2de6b32-2a22-4e1a-a58c-e68cf4817ffe">

**Comment by Hlavtox** [comment]
@MatShir This - `Shop > Currency > Country > Group`, the same in PS settings and on the product itself.

**Comment by kpodemski** [comment]
@Hlavtox best for customer? it never worked like that 🤔

**Comment by MatShir** [comment]
It seems that All Groups overrides everything, I guess it would be the same for country or currency. If you have a specific price with ALL, it will probably be used instead of the define country. It does make sens to ajust. IMO is more a improvement. If we do make the change, she should have an explicit wording

**Comment by MatShir** [comment]
Other question a side: is visitor feature still used ? Since we can really exploit visitors. When the visitor is going to purchase the product he's becoming a customer, so all the specific price will be lost

**Comment by Hlavtox** [comment]
@MatShir That's just default customer groups, used for many purposes. :-) The same scenario will be for any other customer group.

**Comment by MatShir** [comment]
Yes but Visitors has not use, Guest can be exploited, more easily.

**Comment by MatShir** [comment]
related to https://github.com/PrestaShop/PrestaShop/issues/32081

**Comment by Hlavtox** [comment]
@MatShir That issue is interesting and definitely could be a legit scenario, but it has nothing to do with this issue. :-)

Imagine I wrote "Whatevergroupname" instead of "Visitor" in my issue. :D

**Comment by MatShir** [comment]
Yeah I know ! I digressed. I wanted to take this opportunity to get some feedback on your use of visitors. 

To get back to the subject, current behavior: ALL * > specific value

**Comment by MatShir** [comment]
Oh I got it now ! 😓

**Comment by RosaBenouamer** [comment]
The idea is good but must be perhaps flexible. Indeed I imagine sometimes some specific price could be better for visitors to attract them or for groups to thank them. Two alternatives : 
- Give explicitely priority between specific price or 
- Define if the choice of the specific price is driven by the group the most precise or the best price (by a parameter ?)
Your opinion ?
